diff --git a/src/sidebars/generator.ts b/src/sidebars/generator.ts
index 7dc6ece7bedca886c23abe46898e441cafe6761e..fae290b579575cb183146851f2ba048feca312bc 100644
--- a/src/sidebars/generator.ts
+++ b/src/sidebars/generator.ts
@@ -138,7 +138,7 @@ Available doc IDs:
     ): WithPosition<SidebarItemDoc> {
       const {
         sidebarPosition: position,
-        frontMatter: {sidebar_label: label, sidebar_class_name: className},
+        frontMatter: {sidebar_label: label, sidebar_class_name: className, sidebar_custom_props: customProps },
       } = getDoc(id);
       return {
         type: 'doc',
@@ -149,6 +149,7 @@ Available doc IDs:
         // sidebar
         ...(label !== undefined && {label}),
         ...(className !== undefined && {className}),
+        ...(customProps !== undefined && {customProps}),
       };
     }
     function createCategoryItem(
